src/github/config.ts

```diff
@@ -8,7 +11,31 @@ interface RepoInfo {
   path: string;
 }
 
-export async function loadConfigFromGithub(
+export async function getConfig(
```

Comment:
This function contains both GitHub-specific logic (repo name handling), and linter logic (configuration file names). A more extensible design would be like:
- Define a generic filesystem interface (a simple one is okay):
   ```ts
        type LoadFile = (path: string) => Promise<string | null>;
   ```
- Accept this interface to discover configuration (test the supported extensions) in linter:
   ```ts
        function discoverConfig(load: LoadFile): Promise<string>;
   ```
- Define GitHub file source in bot:
   ```ts
        function createGitHubFileSource(apiClient, repoFullName, ref): LoadFile;
   ```
- Wire them up in the bot:
    ```ts
        function lintRepo(apiClient, repoFullName, ref) {
            const loadFile = createGitHubFileSource(apiClient, repoFullName, ref);
            const config = discoverConfig(loadFile);
            // ...
        }
    ```

Later on, when we make the CLI tool, we just need to define a local file source, without duplicating configuration discovery logic.